src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -144,6 +144,12 @@ internal partial class Binder
                         type = BindType(typeSyntax, diagnostics);
                         ParameterHelpers.CheckParameterModifiers(p, diagnostics, parsingFunctionPointerParams: false, parsingLambdaParams: true);
                         refKind = ParameterHelpers.GetModifiers(p.Modifiers, out _, out _, out _, out scope);
+
+                        if (scope == DeclarationScope.Unscoped &&
```

Comment:
It feels like similar logic for determining the ref-scope of out parameters is contained in the `EffectiveScope` accessor. Shouldn't we include this logic in `EffectiveScope` as well?

Reply:
I think this change comes in #63134. We can react to it after this PR.